### PR TITLE
Implement simple JSON index generation

### PR DESCRIPTION
**Pull request by AndreiBarsan**
* Adds tool to dump an index of submap-to-log-chunk to simplify dumping data for 3D reconstruction, neural rendering, etc. experiments.
 * Also improves caching, making it configurable with an environment variable, which includes the ability to completely disable caching, if desired.

### CODE REVIEW

**Comment by AndreiBarsan** [comment]
* **#87** <a href="https://app.graphite.dev/github/pr/pit30m/pit30m/87?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @AndreiBarsan and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/blogpost_00_compression.ipynb
Comment (AndreiBarsan) [inline examples/blogpost_00_compression.ipynb:1]: This is just noisy changes, for some reason it won't let me remove this part from the PR, just ignore the small changes.

File: pit30m/data/submap_indexing.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+_mem = Memory(location=get_pit30m_cache_dir(), verbose=0)
+
+
+@_mem.cache(verbose=0)
+def compute_submap_index(
+    log_ids: list[str],
+    mrp_subsample: int,
+    max_jobs: int = 0,
+) -> dict[str, list[tuple[str, tuple, tuple]]]:
```
Comment (una-dinosauria) [inline pit30m/data/submap_indexing.py:27]: ibid about bultin types for typing

File: pit30m/data/submap_indexing.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    if last_row is not None:
+        chunks.append((cur_submap, tuple(cur_start.tolist()), tuple(last_row.tolist())))
+    return chunks
+
+
+def _load_submap_index_inputs(
+    log_ids: list[str],
+    mrp_subsample: int,
+    max_jobs: int = 0,
+):
```
Comment (una-dinosauria) [inline pit30m/data/submap_indexing.py:121]:
```suggestion
) -> Dict[str, Tuple[List, np.ndarray]]:
```

**Review by una-dinosauria — APPROVED** [review]
Just minor comments

File: pit30m/data/submap_indexing.py (hunk 1 above)
Comment (AndreiBarsan) [reply pit30m/data/submap_indexing.py:27]: Done!

**Pull request merged**